tests/blockchain/eth/contracts/main/worklock/test_worklock.py

```diff
@@ -25,6 +28,54 @@ def token(testerchain, token_economics, deploy_contract):
     return contract
 
 
+@pytest.fixture()
+def router(testerchain, deploy_contract):
+    staking_interface, _ = deploy_contract('StakingInterfaceMock')
+    secret = os.urandom(32)
+    secret_hash = keccak(secret)
+    contract, _ = deploy_contract('StakingInterfaceRouter', staking_interface.address, secret_hash)
+    return contract
+
+
+def next_address(testerchain, worklock):
+    nonce = testerchain.w3.eth.getTransactionCount(worklock.address)
+    if nonce == 0:
+        data = bytes.fromhex('d6') + \
+               bytes.fromhex('94') + \
+               to_canonical_address(worklock.address) + \
+               bytes.fromhex('80')
+    elif nonce <= 127:
+        data = bytes.fromhex('d6') + \
+               bytes.fromhex('94') + \
+               to_canonical_address(worklock.address) + \
+               nonce.to_bytes(1, byteorder='big')
+    elif nonce <= 255:
+        data = bytes.fromhex('d7') + \
+               bytes.fromhex('94') + \
+               to_canonical_address(worklock.address) + \
+               bytes.fromhex('81') + \
+               nonce.to_bytes(1, byteorder='big')
+    elif nonce <= 65535:
+        data = bytes.fromhex('d8') + \
+               bytes.fromhex('94') + \
+               to_canonical_address(worklock.address) + \
+               bytes.fromhex('82') + \
+               nonce.to_bytes(2, byteorder='big')
+    elif nonce <= 16777215:
+        data = bytes.fromhex('d9') + \
+               bytes.fromhex('94') + \
+               to_canonical_address(worklock.address) + \
+               bytes.fromhex('83') + \
+               nonce.to_bytes(3, byteorder='big')
+    else:
+        data = bytes.fromhex('da') + \
+               bytes.fromhex('94') + \
+               to_canonical_address(worklock.address) + \
+               bytes.fromhex('84') + \
+               nonce.to_bytes(4, byteorder='big')
+    return to_checksum_address(keccak(data)[12:32])
```

Comment:
Wow, what's going on here?

Reply:
Calculating future contract address that will be deployed by worklock

Reply:
Can you provide a comment with a link or description of this process?

Reply:
✔️
I simplified calculation and added link to wiki
Also useful links:
* https://ethereum.stackexchange.com/questions/760/how-is-the-address-of-an-ethereum-contract-computed
* https://ethereum.stackexchange.com/questions/24248/how-to-calculate-an-ethereum-contracts-address-during-its-creation-using-the-so